lotus/sem_ops/sem_agg.py

```diff
@@ -181,19 +193,19 @@ def __call__(
             if column not in self._obj.columns:
                 raise ValueError(f"column {column} not found in DataFrame. Given usr instruction: {user_instruction}")
 
-       
-
-
         if group_by:
             grouped = self._obj.groupby(group_by)
-            new_df = pd.DataFrame()
-            for name, group in grouped:
-                res = group.sem_agg(user_instruction, all_cols, suffix, None, progress_bar_desc=progress_bar_desc)
-                new_df = pd.concat([new_df, res])
-            return new_df
-        
-        
-            
+            group_args = [(group, user_instruction, all_cols, suffix, progress_bar_desc) for _, group in grouped]
+            if lotus.settings.enable_multithreading:
+                lotus.logger.debug("Using multithreading")
+                from multiprocessing import Pool
```

Comment:
This implementation spawns a process for each group. What if we spawn threads and keep just one process, since the `process_group` function should not be CPU intensive.

Reply:
Replaced multiprocessing with multithreading